scm/github.go

```diff
@@ -692,3 +692,31 @@ func toRepository(repo *github.Repository) *Repository {
 		Size:    uint64(repo.GetSize()),
 	}
 }
+
+// GetFileContent implements the SCM interface
+func (s *GithubSCM) GetFileContent(ctx context.Context, opt *FileOptions) (string, error) {
+	if !opt.valid() {
+		return "", ErrMissingFields{
+			Method:  "GetFileContent",
+			Message: fmt.Sprintf("%+v", opt),
+		}
+	}
+
+	fileContent, _, _, err := s.client.Repositories.GetContents(ctx, opt.Owner, opt.Repository, opt.Path, nil)
+	if err != nil || fileContent == nil {
```

Comment:
Can `fileContent` be `nil` if `err != nil`??
This would be bad practice to have such an API that allows to returning `nil` for both `err` and `fileContent`

Reply:
Never mind. Checked the API, and it appears to be like this because the fields ignored can be directory content.